zkevm-circuits/src/evm_circuit/util/math_gadget.rs

```diff
@@ -200,6 +201,268 @@ impl<F: FieldExt, const N: usize> AddWordsGadget<F, N> {
+            tmp_radix = tmp_radix * radix_constant_8.clone();
+        }
+
+        //use rangecheck to check v0,v1
+        //we need v0,v1 in range[0,2^66)
+        for idx in 0..9 {
+            cb.range_lookup(v0[idx].expr(), 256);
+            cb.range_lookup(v1[idx].expr(), 256);
+        }
+
+        //radix_constant_64 == 2^64
+        //radix_constant_128 == 2^128
+        let radix_constant = (1u64 << 32).expr();
+        let radix_constant_64 = radix_constant.clone() * radix_constant;
+        let radix_constant_128 =
+            radix_constant_64.clone() * radix_constant_64.clone();
+        cb.require_equal(
+            "mul(multipliers_lo) == product_lo + radix_lo ⋅ 2^128",
+            cur_v0.clone() * radix_constant_128.clone(),
+            t0.expr() + t1.expr() * radix_constant_64.clone()
+                - (c_digits[0].clone()
+                    + c_digits[1].clone() * radix_constant_64.clone()),
+        );
+        cb.require_equal(
+            "mul(multipliers_high) == product_high + radix_high ⋅ 2^128",
+            cur_v1 * radix_constant_128,
+            cur_v0 + t2.expr() + t3.expr() * radix_constant_64.clone()
+                - (c_digits[2].clone()
+                    + c_digits[3].clone() * radix_constant_64),
+        );
+
+        Self {
+            multiplier,
+            multiplicand,
+            product,
+            t0,
+            t1,
+            t2,
+            t3,
+            v0,
+            v1,
+        }
+    }
+
+    pub(crate) fn assign(
+        &self,
+        region: &mut Region<'_, F>,
+        offset: usize,
+        multiplier: Word,
+        multiplicand: Word,
+        product: Word,
+    ) -> Result<(), Error> {
+        self.assign_witness(
+            region,
+            offset,
+            &multiplier,
+            &multiplicand,
+            &product,
+        )?;
+        self.multiplier.assign(
+            region,
+            offset,
+            Some(multiplier.to_le_bytes()),
+        )?;
+        self.multiplicand.assign(
+            region,
+            offset,
+            Some(multiplicand.to_le_bytes()),
+        )?;
+        self.product
+            .assign(region, offset, Some(product.to_le_bytes()))?;
+        Ok(())
+    }
+
+    pub(crate) fn product(&self) -> &util::Word<F> {
+        &self.product
+    }
+
+    //assign t0 ~ t3 and v0, v1
+    fn assign_witness(
+        &self,
+        region: &mut Region<'_, F>,
+        offset: usize,
+        wa: &Word,
+        wb: &Word,
+        wc: &Word,
+    ) -> Result<(), Error> {
+        use num::BigUint;
+
+        let a = BigUint::from_bytes_le(&wa.to_le_bytes());
+        let b = BigUint::from_bytes_le(&wb.to_le_bytes());
+        let c = BigUint::from_bytes_le(&wc.to_le_bytes());
+        let constant_64 = BigUint::from(1u128 << 64);
+        let constant_128 = constant_64.clone() * constant_64.clone();
+        /*let constant_256 = constant_128.clone() * constant_128.clone();
```

Comment:
remove comments

Reply:
constant_256 contributes to the following assert_eq! expression. Though being commented, not sure we or others need it sometime. Maybe it is better to just leave it there?